### PR TITLE
1072 add item performance --- front end 

### LINKED ISSUE #1072: Add item performance -- front-end
Split out from #1043 based on [this discussion](https://github.com/openmsupply/open-msupply/issues/1043#issuecomment-1399645459).

Don't fetch the whole list, only the first 15-20 items, and then do search-as-you-type after that.

With some UX advice from Anisha.

### PR DESCRIPTION
**Pull request by roxy-dao**
Closes #1072 

Have a separate search input for places that don't require stats to load (everywhere except Request Requisition)

### CODE REVIEW

**Comment by CarlosNZ** [comment]
Cool, looks good. I was wondering that since `StockItemSearchInput` and `StockItemSearchInputWithStats` are visually identical with only slightly different queries, if they should be the same component with just a little extra logic to handle the different cases. Cos the two files are almost identical, which seems a bit wasteful to me.

If it ends up being simpler, it might not be. 🤷‍♂️

File: client/packages/system/src/Item/Components/StockItemSearchInputWithStats/StockItemSearchInputWithStats.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import { ItemRowWithStatsFragment } from '../../api/operations.generated';
+
+const ItemOption = styled('li')(({ theme }) => ({
+  color: theme.palette.gray.main,
+  backgroundColor: theme.palette.background.toolbar,
+}));
+
+const filterOptions = {
+  stringify: (item: ItemRowWithStatsFragment) => `${item.code} ${item.name}`,
+  // limit: 100, // unsure why we had the limit, and why so low. performance is ok for me
```
Comment (mark-prins) [inline client/packages/system/src/Item/Components/StockItemSearchInputWithStats/StockItemSearchInputWithStats.tsx:20]: hehe. perhaps time to remove my comment! 😉

**Review by mark-prins — COMMENTED** [review]
cool. only two things to add:
- it's too fast!! I'm using the 3000+ item data file and the item select is super snappy 😁 
- what @CarlosNZ said. can we combine the component code for the two item selection components?

File: client/packages/system/src/Item/Components/StockItemSearchInputWithStats/StockItemSearchInputWithStats.tsx (hunk 1 above)
Comment (roxy-dao) [reply client/packages/system/src/Item/Components/StockItemSearchInputWithStats/StockItemSearchInputWithStats.tsx]: [b766752](https://github.com/openmsupply/open-msupply/pull/1134/commits/b7667522b27727fe7b2d202a5fb0e72e84faaaeb) ~

**Review by mark-prins — APPROVED** [review]
changes look great, thanks @roxy-dao!
I was thinking to have a single `StockItemSearchInput` and have a prop to say if you 'want stats with that'.. but what you've done does make the props neater and removes most of the duplication, so I think that its better overall 😄

**Pull request merged**